code/segmentedtickerplant/pubsub.q

```diff
@@ -15,14 +15,20 @@ subrequestfiltered:([]tbl:`$();handle:`int$();filts:();columns:())
 
 // Function to send end of period messages to subscribers
 // Assumes that .u.endp has been defined on the client side
-endp:{
-  (neg distinct raze union/[value subrequestall;exec handle from .stpps.subrequestfiltered])@\:(`.u.endp;x;y);
- };
+// endp:{
+//   (neg distinct raze union/[value subrequestall;exec handle from subrequestfiltered])@\:(`.u.endp;x;y);
+//  };
 
 // Function to send end of day messages to subscribers      
 // Assumes that .u.end has been defined on the client side   
+// end:{
+//   (neg distinct raze union/[value subrequestall;exec handle from subrequestfiltered])@\:(`.u.end;x);
+//  };
+
+// combine endp and end
+// keep original available until tested
```

Comment:
do you need the commented versions above? 
I think for now it would be better to keep the two version of the function above, you could probably just create a allhandles[] functions that returns 
distinct raze union/[value subrequestall;exec handle from subrequestfiltered]
and is then used by both